### PR TITLE
Api enable epochs endpoint

### PR DESCRIPTION
**Pull request by mateigraura**
Expose activation epochs api:
- save activation epochs metrics 
- add new endpoint route `/network/enable-epochs`

Testing procedure:
- after successful system test, send a GET request to the endpoint `/network/enable-epochs` on any observer
- compare the metrics returned with the activation epochs values from `enableEpochs.toml`

### CODE REVIEW

File: cmd/node/config/api.toml [hunk 1]
```diff
@@ -65,7 +65,10 @@
 
         # /network/config will return metrics related to current configuration of the network (number of shards,
         # consensus group size and so on)
-        { Name = "/config", Open = true }
+        { Name = "/config", Open = true },
+
+        # /network/enable-epochs will return metric related to activation epochs
```
Comment (miiu96) [inline cmd/node/config/api.toml:70]: *metrics

File: statusHandler/statusMetricsProvider.go [hunk 2]
```diff
@@ -183,6 +183,34 @@ func (sm *statusMetrics) ConfigMetrics() map[string]interface{} {
 	return configMetrics
 }
 
+// EnableEpochMetrics will return metric related to activation epochs
```
Comment (miiu96) [inline statusHandler/statusMetricsProvider.go:186]: metrics

File: node/external/interface.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ type StatusMetricsHandler interface {
 	StatusMetricsWithoutP2PPrometheusString() string
 	EconomicsMetrics() map[string]interface{}
 	ConfigMetrics() map[string]interface{}
+	EnableEpochMetrics() map[string]interface{}
```
Comment (miiu96) [inline node/external/interface.go:24]: EnableEpochsMetrics

File: integrationTests/mock/statusMetricsStub.go [hunk 4]
```diff
@@ -24,6 +25,11 @@ func (sms *StatusMetricsStub) ConfigMetrics() map[string]interface{} {
 	return sms.ConfigMetricsCalled()
 }
 
+// EnableEpochMetrics
+func (sms *StatusMetricsStub) EnableEpochMetrics() map[string]interface{} {
```
Comment (miiu96) [inline integrationTests/mock/statusMetricsStub.go:29]: EnableEpochsMetrics

File: core/constants.go [hunk 5]
```diff
@@ -397,6 +397,73 @@ const AsyncCallStepField = "AsyncCallStep"
 // before executing the destination async call, to be put aside for the async callback
 const AsyncCallbackGasLockField = "AsyncCallbackGasLock"
 
+const (
+	// MetricScDeployEpoch represents the epoch when the deployment of smart contracts will be enabled
+	MetricScDeployEpoch = "smart_contract_deploy"
```
Comment (bogdan-rosianu) [inline core/constants.go:402]: I would recommend the suffix `EnableEpoch` instead of `Epoch`

Comment (bogdan-rosianu) [reply core/constants.go:402]: also all the metrics should have the prefix `erd_`. So:

`erd_smart_contract_deploy_enable_epoch`
apply to all the metrics

File: facade/disabled/disabledStatusMetrics.go [hunk 6]
```diff
@@ -37,6 +37,10 @@ func (d *disabledStatusMetricsHandler) ConfigMetrics() map[string]interface{} {
 	return getReturnMap()
 }
 
+func (d *disabledStatusMetricsHandler) EnableEpochMetrics() map[string]interface{} {
```
Comment (bogdan-rosianu) [inline facade/disabled/disabledStatusMetrics.go:40]: missing comment

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: cmd/node/config/api.toml (hunk 1 above)
Comment (mateigraura) [reply cmd/node/config/api.toml]: done

File: statusHandler/statusMetricsProvider.go (hunk 2 above)
Comment (mateigraura) [reply statusHandler/statusMetricsProvider.go]: done

File: node/external/interface.go (hunk 3 above)
Comment (mateigraura) [reply node/external/interface.go]: done

File: integrationTests/mock/statusMetricsStub.go (hunk 4 above)
Comment (mateigraura) [reply integrationTests/mock/statusMetricsStub.go]: done

File: core/constants.go (hunk 5 above)
Comment (mateigraura) [reply core/constants.go]: done

File: facade/disabled/disabledStatusMetrics.go (hunk 6 above)
Comment (mateigraura) [reply facade/disabled/disabledStatusMetrics.go]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by miiu96 — DISMISSED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by gabi-vuls — APPROVED** [review]
System test passed

**Review by miiu96 — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**